File: Payhub - SDK/php_SDK/com/payhub/ws/api/RefundInformation.php
<?php

/**
 * Created by PhpStorm.
 * User: agustin
 * Date: 24/07/2015
 * Time: 10:53
 */
class RefundInformation
{
    public $transaction_id;
    public $lastRefundResponse;
    public $_links;
    public $errors;
    public $rowData;
    public $metaData;
    private $transactionManager;
    private $merchantInformation;
    public $settlementStatus;
    public $merchantOrganizationId;

    /**
     * @return mixed
     */
    public function getSettlementStatus()
    {
        return $this->settlementStatus;
    }

    /**
     * @param mixed $settlementStatus
     */
    public function setSettlementStatus($settlementStatus)
    {
        $this->settlementStatus = $settlementStatus;
    }

    /**
     * RefundInformation constructor.
     * @param $transactionManager
     */
    public function __construct()
    {
        $this->transactionType=TransactionType::Refund;
    }

    /**
     * @return mixed
     */
    public function getTransactionId()
    {
        return $this->transaction_id;
    }

    /**
     * @param mixed $transaction_id
     */
    public function setTransactionId($transaction_id)
    {
        $this->transaction_id = $transaction_id;
    }

    /**
     * @return mixed
     */
    public function getLastRefundResponse()
    {
        return $this->lastRefundResponse;
    }

    /**
     * @param mixed $lastRefundResponse
     */
    public function setLastRefundResponse($lastRefundResponse)
    {
        $this->lastRefundResponse = $lastRefundResponse;
    }

    /**
     * @return mixed
     */
    public function getMerchantOrganizationId()
    {
        return $this->merchantOrganizationId;
    }

    /**
     * @param mixed $merchantOrganizationId
     */
    public function setMerchantOrganizationId($merchantOrganizationId)
    {
        $this->merchantOrganizationId = $merchantOrganizationId;
    }

    /**
     * @return mixed
     */
    public function getLinks()
    {
        return $this->_links;
    }

    /**
     * @param mixed $links
     */
    public function setLinks($links)
    {
        $this->_links = $links;
    }

    /**
     * @return mixed
     */
    public function getErrors()
    {
        return $this->errors;
    }

    /**
     * @param mixed $errors
     */
    public function setErrors($errors)
    {
        $this->errors = $errors;
    }

    /**
     * @return mixed
     */
    public function getRowData()
    {
        return $this->rowData;
    }

    /**
     * @param mixed $rowData
     */
    public function setRowData($rowData)
    {
        $this->rowData = $rowData;
    }

    /**
     * @return mixed
     */
    public function getMetaData()
    {
        return $this->metaData;
    }

    /**
     * @param mixed $metaData
     */
    public function setMetaData($metaData)
    {
        $this->metaData = $metaData;
    }

    /**
     * @param $transactionManager
     */
    public function setTransactionManager($transactionManager)
    {
        $this->transactionManager = $transactionManager;
    }

    /**
     * @return mixed
     */
    public function getMerchantInformation()
    {
        if($this->merchantInformation==null){
            $m = new MerchantInformation($this->transactionManager);
            $m->getDataByTransaction(TransactionType::Refund, $this->lastRefundResponse->getRefundTransactionId());
            $this->merchantInformation=$m;
        }
        return $this->merchantInformation;
    }
    public static function fromArray($data){
        $refund = new RefundInformation();
        foreach ($data as $key => $value){
            if( property_exists( get_class($refund), $key ) ) {
                if($key=="errors"){
                    $refund->{$key}=Errors::fromArray($value);
                }
                if($key=="lastRefundResponse"){
                    $refund->{$key}=RefundResponse::fromArray($value);
                }
                else{
                    $refund->{$key} = $value;
                }
            }
        }
        return $refund;
    }
}

File: Payhub - SDK/java_SDK/PayHubSDK/src/main/java/com/payhub/ws/api/RefundInformation.java
package com.payhub.ws.api;

import java.io.IOException;
import java.util.List;

import com.payhub.ws.model.Merchant;
import com.payhub.ws.model.RefundResponse;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.core.JsonParseException;
import com.fasterxml.jackson.databind.JsonMappingException;

@JsonInclude(JsonInclude.Include.NON_NULL)  
public class RefundInformation {

    private Merchant merchant;
    private String transaction_id;
    private RefundResponse lastRefundResponse;
    private String merchantOrganizationId;
    public String settlementStatus;
    public Object _links;
    private List<Errors> errors;
    public String rowData;
    private Object metaData;
    private TransactionManager transactionManager;
    private MerchantInformation merchantInformation; 
    
	public String getMetaData() {
		return (String) metaData;
	}
	public void setMetaData(Object metadata) {
		this.metaData = metadata;
	}
	public Merchant getMerchant() {
		return merchant;
	}
	public void setMerchant(Merchant merchant) {
		this.merchant = merchant;
	}
	public String getTransaction_id() {
		return transaction_id;
	}
	public void setTransaction_id(String transaction_id) {
		this.transaction_id = transaction_id;
	}
	public RefundResponse getLastRefundResponse() {
		return lastRefundResponse;
	}
	public void setLastRefundResponse(RefundResponse lastRefundResponse) {
		this.lastRefundResponse = lastRefundResponse;
	}
	public String getMerchantOrganizationId() {
		return merchantOrganizationId;
	}
	public void setMerchantOrganizationId(String merchantOrganizationId) {
		this.merchantOrganizationId = merchantOrganizationId;
	}
	public Object get_links() {
		return _links;
	}
	public void set_links(Object _links) {
		this._links = _links;
	}
	public List<Errors> getErrors() {
		return errors;
	}
	public void setErrors(List<Errors> errors) {
		this.errors = errors;
	}
	public String getRowData() {
		return rowData;
	}
	public void setRowData(String rowData) {
		this.rowData = rowData;
	}
	/**
	 * @param transactionManager the transactionManager to set
	 */
	public void setTransactionManager(TransactionManager transactionManager) {
		this.transactionManager = transactionManager;
	}
	
	/**
	 * @return the merchantInformation
	 * @throws IOException 
	 * @throws JsonMappingException 
	 * @throws JsonParseException 
	 */
	public MerchantInformation getMerchantInformation() throws JsonParseException, JsonMappingException, IOException {
		if(merchantInformation==null){			
				MerchantInformation m = new MerchantInformation(this.transactionManager);
				m.getDataByTransaction(TransactionType.Refund, transaction_id);
				merchantInformation=m;				
			}
		return merchantInformation;
	}
}
